### DISCUSSION TITLE
Build own kernel modules? Need matching linux-headers package

### DISCUSSION THREAD

**Discussion by steve8x8** [OpenNeptune3D/OpenNept4une#161] [Q&A]
I'm running the 0.15 version, and there's a divergence between the kernel itself and the headers that would be required to build extra drivers:
```
# dpkg -l | grep linux-.*-rockchip64
hi  linux-dtb-current-rockchip64      24.2.0-trunk                       arm64        Armbian Linux current DTBs in /boot/dtb-6.1.67-current-rockchip64
ii  linux-headers-current-rockchip64  24.2.1                             arm64        Armbian Linux current headers 6.6.16-current-rockchip64
hi  linux-image-current-rockchip64    24.2.0-trunk                       arm64        Armbian Linux current kernel image 6.1.67-current-rockchip64
```
Where can I find the 24.2.0-trunk `linux-headers-current-rockchip64` package that would match the (held, and therefore immutable) `linux-image` package?

**Comment by poiuztreq** [comment]
https://github.com/OpenNeptune3D/Armbian-ZNP-K1-build

Maybe this will help

**Reply by steve8x8** [reply]
Thanks for the pointer, but I'm not into building a whole new firmware, I'm just looking for the source of the `linux-image` package (which should have a `linux-headers` package next to it, with the matching version). Of course I could lie at `dkms` by adding a symlink, but I'd very probably ask for trouble (24.2.1 doesn't seem too far from 24.2.0 but 6.6.16 is certainly something else than 6.1.67).
If I was to build a new FW I'd go for 6.6.x, but I'm sure I'd ruin more than just the printer by attempting to upgrade the currently held packages

**Reply by halfmanbear** [reply]
I don't believe I retained the compiled headers for that release. The current image has the headers included in the image

**Reply by poiuztreq** [reply]
I compiled the image with EXTRAWIFI=no, and I've provided the .deb files for version 24.5.0 and 6.7.12-edge.

Please ensure to install libelf-dev, as it's a dependency for one of the newer versions and remove armbian-bsp-cli-znp-k1-current or you'll encounter errors.

After installing the .deb files, open OpenNeptune CLI and reinstall your .dtb file.

sudo apt-mark hold linux-image-current-rockchip64
sudo apt-mark hold linux-dtb-current-rockchip64

You can find the .deb files here: https://github.com/poiuztreq/neptune4-bookworm-upgrade/releases/tag/24.5.0_6.7.12
